File: Javascript/KthSmallestElementInBST.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0:val
    this.left = (left === undefined) ? null:left
    this.right = (right === undefined) ? null:right
}

//Iterative Method
var kthSmallestElementBST = function(root){
    let n = 0
    const stack = []
    let cur = root

    while(cur !== null || stack){
        while(cur){
            stack.push(cur)
            cur = cur.left
        }
        cur = stack.pop()
        n+=1
        if(n===k) return cur.val
        cur = cur.right
    }
}

File: Python/KthSmallestElementInBST.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right

##Recursion
class Solution:
    def kthSmallestElementBST(root):

        array = []

        def dfs(node):
            nonlocal array

            if not node:
                return
            
            dfs(node.left)
            array.append(node.val)
            dfs(node.right)

        dfs(root)

        return array[k-1]
